Update SlicerAndPrintStart.md and add sensorless writeup to tuning (#418)
VoronDesign/Voron-Documentation: tuning/sensorless.md

---
layout: default
title: Configuring sensorless homing
parent: Tuning Guides
nav_exclude: true
---

This writeup covers how to set up sensorless homing on an Octopus 1.1 with 2209 drivers. Different drivers and/or MCUs may differ. Adjust accordingly.

Also note that there are example working configurations at the bottom of this guide. Use them as examples based on your MCU and drivers, etc.

{: .note }

> IF YOU DO NOT READ THIS CAREFULLY AND ADJUST THINGS WHEN NEEDED YOU WILL BE SAD. IF YOU USE A PHYSICAL ENDSTOP YOU MUST CHANGE THINGS TO HOME Z TO THE ENDSTOP PIN, NOT THE MIDDLE OF THE BED. YOU HAVE BEEN WARNED!

Reviewer: Not a blocker for merge but maybe making it clearer that the user will need to update macros, specifically, would be useful?
```suggestion
IF YOU DO NOT READ THIS CAREFULLY AND ADJUST THINGS WHEN NEEDED YOU WILL BE SAD. IF YOU USE A PHYSICAL ENDSTOP YOU MUST CHANGE MACROS TO HOME Z TO THE ENDSTOP PIN, NOT THE MIDDLE OF THE BED. YOU HAVE BEEN WARNED!
```

Author: i agree. its implied in there but this sets them up for success better

At the end of this document is a section with working examples you can use as a reference for some common hardware

# Getting started

Since this example revolves around an Octopus 1.1, you need to verify which diag jumpers to install based on your hardware and motor positions. The examples below serve to illustrate the process.

1. Power off the printer gracefully
2. Physically disconnect the X and Y endstop connections from the MCU.

> 3. Plug in jumpers on the `DIAG` ports as shown below (**J16** and **J17**). We use these ports since thats the corresponding diag jumper for where our XY steppers are connected.

Reviewer: ```suggestion
3. Plug in jumpers on the `DIAG` ports as shown below (**J16** and **J17**). We use these ports since these are the corresponding diag jumpers for where our XY steppers are connected.
```

3. Edit `homing.cfg` 
4. **Add** the macro from step 7 to `homing.cfg`. **_Be sure to edit the coordinates as needed for your bed in the Z section (the line that reads **G1 X175 Y175 F15000**). IE to the center of the bed, your endstop pin for z, etc._**
5. Save `homing.cfg` 
6. Edit your `printer.cfg` file
7. If you have a `[safe_z_home]` section, find it and comment it out as we will be using homing override as found in `homing.cfg`.

The `[homing_override]` block we have in the above macros is now going to be responsible for all homing, whether via the buttons in Mainsail or Fluidd, or via commands like G28 X, etc. This override allows us, the end user, to customize how the homing operation happens. When using sensorless, this is important, as it lets us adjust things like the current used for homing, etc. More on this later.

### Danger Klipper

While it sounds dangerous, there is nothing to be afraid of! Choosing this route means you do not need to do any of the macro stuff as listed above, as we can just simply add a new property in our X and Y driver section to handle homing current. Thats it!

> If you want to switch to Danger Klipper, now is the time to do so. You can follow [this](https://github.com/EricZimmerman/VoronTools/blob/main/DangerKlipper.md) to make the switch! Switching will not make you change anything else about your configutation.

Reviewer: - links should have a bigger tap target for users on mobile than "this"
- configutation -> configuration
```suggestion
If you want to switch to Danger Klipper, now is the time to do so. You can [follow this other guide](https://github.com/EricZimmerman/VoronTools/blob/main/DangerKlipper.md) to make the switch! Switching will not make you change anything else about your configuration.
```

## Updating stepper_y

1. Locate the `[stepper_y]` section
2. Record the current value for `endstop_pin` (**PG9** for example)
3. **Change** the `endstop_pin` to `tmc2209_stepper_y:virtual_endstop`
4. **Change** `homing_speed` to `80`
5. **Change** `homing_retract_dist` to `0`
6. Locate the `[tmc2209 stepper_y]` section (usually right below where you just edited)
7. **Add** `diag_pin` to match what you recorded from step 2, but add a ^ before it.
    Example: `diag_pin: ^PG9`
8. **Add** this below the `diag_pin` entry: `driver_SGTHRS: 255`

For those of you that took the Danger Klipper route above, add this line below your `run_current` property:

```
home_current: 0.49
current_change_dwell_time: 0.2
```

With those changes complete, save your configuration.

## Other changes

1. Move to the top of your `printer.cfg`. If not already present, add the following configuration section:
    ```
    [force_move]
    enable_force_move: True
    ```
2. If you went with stock Klipper, add an include for wherever you created your `homing.cfg` file. Example: `[include homing.cfg]`
3. Save your `printer.cfg` file and allow Klipper to restart.

## Finding the appropriate driver value

> Recall earlier we set the driver threshhold to 255. This is the most sensitive value and will almost certainly NOT work for your use case. We now need to find the appropriate value, or rather, range of values that will work for you.

Reviewer: ```suggestion
Recall earlier we set the driver threshold to 255. This is the most sensitive value and will almost certainly NOT work for your use case. We now need to find the appropriate value, or rather, range of values that will work for you.
```

    SET_TMC_CURRENT STEPPER=stepper_y CURRENT={RUN_CURRENT_Y}
```
Finally, restore our motor's run current we captured before. 

### Danger Klipper details

Nothing to see here. Danger Klipper manages the process

## Finding the right driver value

When you first start trying to find your value, because of the macro above, it will look like the toolhead is moving away from the right rail, then stopping (for the stock Klipper folks). 

This is normal and just means the value is still too sensitive to home properly.

Early on, I tend to jump down in jumps of 50. At some point you will get X to home all the way to the rail. 

However, if you went TOO low, it might bump harder into the rail than it should. In this case, ADD half the value you last went down by and repeat steps 1 and 2. For non 2209 people, you would of course be increasing from -64, so adjust accordingly.

Eventually you will find the BIGGEST number that homes X successfully. Nice!

With the maximum found, continue to DECREASE the value by 5 or so until X homes, but bumps too hard into the rail. You may need to walk this in by changing the value by 1 when getting close.

> This is your MINIUMUM value.

Reviewer: ```suggestion
This is your MINIMUM value.
```

---
layout: default
title: Configuring sensorless homing
parent: Tuning Guides
nav_exclude: true
---

This writeup covers how to set up sensorless homing on an Octopus 1.1 with 2209 drivers. Different drivers and/or MCUs may differ. Adjust accordingly.

Also note that there are example working configurations at the bottom of this guide. Use them as examples based on your MCU and drivers, etc.

{: .note }
IF YOU DO NOT READ THIS CAREFULLY AND ADJUST THINGS WHEN NEEDED YOU WILL BE SAD. IF YOU USE A PHYSICAL ENDSTOP YOU MUST CHANGE THINGS TO HOME Z TO THE ENDSTOP PIN, NOT THE MIDDLE OF THE BED. YOU HAVE BEEN WARNED!

> At the end of this document is a section with working examples you can use as a reference for some common hardware

Reviewer: Probably don't need to mention this twice.

Author: removed

# Getting started

> As noted earlier, you need to verify which diag jumpers to install based on your hardware and motor positions! The examples below serve to illustrate the process.

Reviewer: When was this noted earlier?

Author: adjusted
